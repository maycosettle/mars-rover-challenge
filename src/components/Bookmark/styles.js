const useStyles = () => ({
  iconContainer: 'mx-auto flex items-center justify-center h-12 w-12 rounded-full bg-yellow-100',
  icon: 'h-6 w-6 text-yellow-600',
  contentContainer: 'mt-3 text-center sm:mt-5',
  title: 'text-lg leading-6 font-medium text-gray-900',
  inputContainer: 'mt-2',
  label: 'text-sm text-gray-500',
  input: 'flex items-center justify-center w-1/2 p-2 m-auto',
});

export default useStyles;
