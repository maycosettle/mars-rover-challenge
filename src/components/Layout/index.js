export * from './PageLayout';
export * from './PaperLayout';
export * from './PhotoLayout';
