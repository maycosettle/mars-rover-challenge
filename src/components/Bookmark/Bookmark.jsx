import PropTypes from 'prop-types';
import { Dialog } from '@headlessui/react';
import { ExclamationIcon } from '@heroicons/react/solid';

import { Input } from 'components/Input';

import useStyles from './styles';

const Bookmark = ({ onChange }) => {
  const styles = useStyles();

  return (
    <>
      <div className={styles.iconContainer}>
        <ExclamationIcon className={styles.icon} aria-hidden="true" />
      </div>
      <div className={styles.contentContainer}>
        <Dialog.Title as="h3" className={styles.title}>
          Do you want to save the bookmark?
        </Dialog.Title>
        <div className={styles.inputContainer}>
          <p className={styles.label}>Enter a name for the bookmark</p>
          <div className={styles.input}>
            <Input onChange={onChange} />
          </div>
        </div>
      </div>
    </>
  );
};

Bookmark.propTypes = {
  onChange: PropTypes.func,
};

Bookmark.defaultProps = {
  onChange: () => {},
};

export default Bookmark;
