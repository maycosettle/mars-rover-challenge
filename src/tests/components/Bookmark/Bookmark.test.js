import { shallow } from 'enzyme';
import { Bookmark } from 'components/Bookmark';

describe('tests on <Bookmark />', () => {
  const wrapper = shallow(<Bookmark />);

  test('Should render Bookmark correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
