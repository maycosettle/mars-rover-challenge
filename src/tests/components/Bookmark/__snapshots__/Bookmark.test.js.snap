// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <Bookmark /> Should render Bookmark correctly 1`] = `
<Fragment>
  <div
    className="mx-auto flex items-center justify-center h-12 w-12 rounded-full bg-yellow-100"
  >
    <ExclamationIcon
      aria-hidden="true"
      className="h-6 w-6 text-yellow-600"
    />
  </div>
  <div
    className="mt-3 text-center sm:mt-5"
  >
    <Title
      as="h3"
      className="text-lg leading-6 font-medium text-gray-900"
    >
      Do you want to save the bookmark?
    </Title>
    <div
      className="mt-2"
    >
      <p
        className="text-sm text-gray-500"
      >
        Enter a name for the bookmark
      </p>
      <div
        className="flex items-center justify-center w-1/2 p-2 m-auto"
      >
        <Input
          id=""
          label=""
          name=""
          onChange={[Function]}
          placeholder=""
        />
      </div>
    </div>
  </div>
</Fragment>
`;
